app/middlewares/token-validation.js

```diff
@@ -0,0 +1,13 @@
+const jwt = require('jsonwebtoken');
+const { validationError, INVALID_TOKEN } = require('../errors');
+
+const tokenValidation = (req, _res, next) => {
+  jwt.verify(req.headers.authorization, process.env.TOKEN_KEY, err => {
```

Comment:
Use `verifyAsync`

Reply:
There's no function called `verifyAsync` in jsonwebtoken library, but `verify` is asynchronous when a callback is supplied, i.e. in this case `verify` runs asynchronously.